llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -2799,14 +2807,15 @@ template <> Value *SPIRVToLLVM::transValueWithOpcode<OpAccessChain>(SPIRVValue *
 
     // Do the final index if we have one.
     if (spvAccessChain->isInBounds())
-      return getBuilder()->CreateInBoundsGEP(newBase, indices);
+      return getBuilder()->CreateInBoundsGEP(newBase->getType()->getScalarType()->getPointerElementType(), newBase,
+                                             indices);
     else
-      return getBuilder()->CreateGEP(newBase, indices);
+      return getBuilder()->CreateGEP(newBase->getType()->getScalarType()->getPointerElementType(), newBase, indices);
   } else {
     if (spvAccessChain->isInBounds())
-      return getBuilder()->CreateInBoundsGEP(base, indices);
+      return getBuilder()->CreateInBoundsGEP(base->getType()->getScalarType()->getPointerElementType(), base, indices);
     else
-      return getBuilder()->CreateGEP(base, indices);
+      return getBuilder()->CreateGEP(base->getType()->getScalarType()->getPointerElementType(), base, indices);
```

Comment:
Could you hoist `base`/`newBase->getType()->get()ScalarType()->getPointerElementType()` into local variables to avoid these long expressions?